software/android_app/app/src/main/java/com/example/signbuddy/LessonActivity.java

```diff
@@ -1,47 +1,235 @@
+        String alphaNumber = String.valueOf(alphanum);
+        String fnm = "alpha_"+alphaNumber; //  image file name
+        String PACKAGE_NAME = getApplicationContext().getPackageName();
+        int imgId = getResources().getIdentifier(PACKAGE_NAME+":drawable/"+fnm , null, null);
+        image.setImageResource(imgId);
+
+        //start countdown
+        new CountDownTimer(7500, 1500) {
+            public void onTick(long millisUntilFinished) {
+                countText = findViewById(R.id.countText);
+                countText.setText("Perform Sign in: " + ((millisUntilFinished / 1500)+1));
+            }
+            public void onFinish() {
+                countText.setText("GO!");
+                BleManager.getInstance().notify(MainActivity.SignBuddy, UUID_SERVICE, UUID_CHARACTERISTIC_NOTIFY, new BleNotifyCallback() {
+                    @Override
+                    public void onNotifySuccess() {
+
+                    }
+
+                    @Override
+                    public void onNotifyFailure(BleException exception) {
+
+                    }
+
+                    @Override
+                    public void onCharacteristicChanged(byte[] data) {
+                        for (byte b : data) {
+                            if (b == MSG_SYNC && MainActivity.byteBuffer.size() == 0) {
+                                skip_crc = 4;
+                                next_byte_is_length = true;
+                            } else if (next_byte_is_length) {
+                                msg_length = b;
+                                next_byte_is_length = false;
+                            } else if (skip_crc > 0) {
+                                skip_crc--;
+                            } else {
+                                MainActivity.byteBuffer.write(b);
+                                if (MainActivity.byteBuffer.size() == msg_length) {
+                                    new Thread(new LessonActivity.ParseMessage(MainActivity.byteBuffer.toByteArray())).start();
+                                    MainActivity.byteBuffer.reset();
+                                }
+                            }
+                        }
+                    }
+                });
+                countText.setText(gestureData.toString());
+
+            }
+        }.start();
+
+        if (checkCorrect()==true) {
+            Toast.makeText(LessonActivity.this, "Correct!", Toast.LENGTH_SHORT).show();
+            alpha++;
+            finish();
+            i.addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION);
+            Bundle mbundle = new Bundle();
+            mbundle.putInt("alpha", alpha);
+            mbundle.putString("LessonNumber", lessonNumber);
+            i.putExtras(mbundle);
+
+            startActivity(i);
+            overridePendingTransition(0, 0);
+        }
+        else {
+            Toast.makeText(LessonActivity.this, "Incorrect, Please try again!", Toast.LENGTH_SHORT).show();
```

Comment:
I think toast is mostly used for debug info, I think these important messages should be displayed as text in the activity screen